python/py-qdarkstyle/Portfile

```diff
@@ -24,13 +24,16 @@ checksums           sha256  6a967c4b664446f8bed9df12d1032cf68cb54f186bfc9cbfdbbc
                     rmd160  95b7cbcdbec958e2ee685aff8348e5601a2a1c35 \
                     size    405579
 
-python.versions     27 35 36 37
+python.versions     27 35 36 37 38
 
 if {${name} ne ${subport}} {
+    depends_build-append \
+                    port:py${python.version}-setuptools
+
     depends_lib-append \
                     port:py${python.version}-helpdev \
                     port:py${python.version}-qtpy \
-                    port:py${python.version}-setuptools
+                    port:py${python.version}-qtsass
```

Comment:
I don't agree with the change in dependency type. The port installs a ```console_script```, which uses ```pkg_resources``` from ```py-setuptools``` and it is, therefore, also required at runtime. 

Secondly, the ```py-qtsass``` is listed as under ```extra_requires``` in the category "develop", which I typically wouldn't add. However, I do  agree that the unconditional import in the file you mention would suggest it should not be listed as an optional dependency... If you do include this dependency, what is the rationale for not adding ```watchdog```, which is also imported in ```scripts/process_qrc.py```?

Reply:
> The port installs a `console_script`, which uses `pkg_resources` from `py-setuptools` and it is, therefore, also required at runtime.

Please forgive my ignorance, but would you mind telling me where the `console_script` is?
I am not disagreeing with your assessment, but I am unable to find the installed `console_script`, and I would like to expand my own knowledge.

> [W]hat is the rationale for not adding watchdog, which is also imported in scripts/process_qrc.py?

Again, I do not mean to be obtuse, but I cannot find a file called `process_qrc.py`.

Reply:
ah, sorry you are right about the ```process_qrc.py``` file, which is present in the GitHub repository, but apparently does not get installed. So then it should be fine to only add ```py-qtsass```. 

Regarding the console_script, it's this line in the setup.py:
```    entry_points={"console_scripts": ["qdarkstyle=qdarkstyle.__main__:main"]},```
which installs for py37-qdarkstyle the file ```/opt/local/bin/qdarkstyle-3.7```. If you look at its contents you'll notice that it does ```from pkg_resources import load_entry_point```, which will require ```py-setuptools``` at runtime.

Reply:
Thank you for taking the time for the addition explanation.

Reply:
Hi, if this really is a hard dependency, (1) I would suggest to have this fixed upstream and (2) just add it as `depends_lib`.